### PR TITLE
feat(id-modal): make BaseIdModalOpen rendering configurable

### PR DESCRIPTION
**Pull request by CachedaCodes**
[EX-6751](https://searchbroker.atlassian.net/browse/EX-6751)

## Motivation and context
Replicating the changes from `BaseIdModalClose` in `BaseIdModalOpen`

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?
Create a new `BaseIdModalOpen` and see the new slot features. Also, a new test for the new named slot was created.

### CODE REVIEW

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
-  </BaseEventButton>
+  <NoElement v-on="$listeners" data-test="open-modal-id">
+    <!--
+      @slot opening-element. It's the element that will trigger the modal opening. It's a
+      button by default.
+        @binding {Function} openModal - The function to open the modal.
+    -->
+    <slot :openModal="emitOpenModalEvent" name="opening-element">
+      <button @click="emitOpenModalEvent" class="x-button x-events-modal-id-open-button">
+        <slot />
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:10]: Add slot documentation as you did un the `BaseIdModalClose` component

```suggestion
        <!-- @slot (Required) Button content with a text, an icon or both -->
        <slot />
```

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 2]
```diff
@@ -66,6 +70,35 @@ The component rendering content passed to the default slot and opening the modal
...
+  export default {
+    name: 'BaseIdModalOpenTest',
+    components: {
+      BaseIdModalOpen
+    }
+  };
+</script>
+```
+
 ## Events
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:102]: Can you move this section to the top of the `docs`? Same in the `base-id-modal-close` file.

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
good job!

File: packages/x-components/src/components/modals/base-id-modal-open.vue (hunk 1 above)
Comment (CachedaCodes) [reply packages/x-components/src/components/modals/base-id-modal-open.vue:10]: 🤦 How did I mess this up?

File: packages/x-components/src/components/modals/base-id-modal-open.vue (hunk 2 above)
Comment (CachedaCodes) [reply packages/x-components/src/components/modals/base-id-modal-open.vue:102]: Not sure if I follow, I see them at the same height both in this file and in `base-id-modal-close`.

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
   import { Component, Prop } from 'vue-property-decorator';
-  import { XEventsTypes } from '../../wiring/events.types';
-  import BaseEventButton from '../base-event-button.vue';
+  import { NoElement } from '../no-element';
 
   /**
-   * Component containing an event button that emits {@link XEventsTypes.UserClickedOpenModal} when
-   * clicked with the modalId as payload. It has a default slot to customize its contents.
+   * Component that allows to open a modal by emitting {@link XEventsTypes.UserClickedOpenModal}
+   * with the modalId as payload. It's fully customizable as it exposes the opening event but by
```
Comment (javieri-empathy) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:23]: I'm not sure that this sentece makes much sense, probably you meant the slot?

> It's fully customizable as it exposes the opening event

File: packages/x-components/src/components/modals/base-id-modal-open.vue (hunk 1 above)
Comment (javieri-empathy) [reply packages/x-components/src/components/modals/base-id-modal-open.vue:10]: This is still not done

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 4]
```diff
@@ -1,36 +1,39 @@
...
+    components: { NoElement }
   })
   export default class BaseIdModalOpen extends Vue {
     @Prop({ required: true })
     protected modalId!: string;
 
-    protected get events(): Partial<XEventsTypes> {
-      return { UserClickedOpenModal: this.modalId };
+    protected emitOpenModalEvent(): void {
+      this.$x.emit('UserClickedOpenModal', this.modalId, { target: this.$el as HTMLElement });
```
Comment (javieri-empathy) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:36]: Not really relevant, as we don't use it, but should we use `event.target` as the `target` metadata field here? If you replace the template, the target of the even't might not be the wrapper.

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 5]
```diff
@@ -1,36 +1,39 @@
...
-    components: { BaseEventButton }
+    components: { NoElement }
   })
   export default class BaseIdModalOpen extends Vue {
     @Prop({ required: true })
     protected modalId!: string;
 
-    protected get events(): Partial<XEventsTypes> {
-      return { UserClickedOpenModal: this.modalId };
+    protected emitOpenModalEvent(): void {
```
Comment (javieri-empathy) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:35]: Shouldn't this properties be documented? Also the `modalId` is part of the public API.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/modals/base-id-modal-open.vue (hunk 2 above)
Comment (LuisMartinez15) [reply packages/x-components/src/components/modals/base-id-modal-open.vue:102]: Yep, in both files this section should be at the top of the `docs`. See [HistoryQueriesSwitch](https://github.com/empathyco/x/blob/099121b27b08cb898652045661fbf92f81e1024e/packages/x-components/src/x-modules/history-queries/components/history-queries-switch.vue) for reference.

Comment (CachedaCodes) [reply packages/x-components/src/components/modals/base-id-modal-open.vue:118]: Ah Ok, got you.

File: packages/x-components/src/components/modals/base-id-modal-close.vue [hunk 6]
```diff
@@ -16,30 +21,47 @@
...
+     *
+     * @public
+     */
     @Prop({ required: true })
     protected modalId!: string;
 
-    protected emitCloseModalEvent(): void {
-      this.$x.emit('UserClickedCloseModal', this.modalId, { target: this.$el as HTMLElement });
+    /**
+     * Emits the 'UserClickedCloseModal' event with the modalId as payload.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/base-id-modal-close.vue:46]:
```suggestion
     * Emits the {@link XEventTypes.UserClickedCloseModal} event with the modalId as payload.
```

File: packages/x-components/src/components/modals/base-id-modal-open.vue [hunk 7]
```diff
@@ -1,50 +1,77 @@
...
+     *
+     * @public
+     */
     @Prop({ required: true })
     protected modalId!: string;
 
-    protected get events(): Partial<XEventsTypes> {
-      return { UserClickedOpenModal: this.modalId };
+    /**
+     * Emits the 'UserClickedOpenModal' event with the modalId as payload.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/base-id-modal-open.vue:46]:
```suggestion
     * Emits the {@link XEventTypes.UserClickedOpenModal} event with the modalId as payload.
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

**Review by LuisMartinez15 — DISMISSED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**